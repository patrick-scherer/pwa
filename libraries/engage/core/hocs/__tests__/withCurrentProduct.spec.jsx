import React from 'react';
import { mount } from 'enzyme';
import { withCurrentProduct } from '../withCurrentProduct';

jest.mock('@shopgate/pwa-common/context', () => {
  // eslint-disable-next-line global-require
  const { createContext } = require('react');

  return {
    ThemeContext: createContext({
      contexts: {
        ProductContext: createContext({
          id: '123',
          name: 'acme',
          price: {
            currency: 'EUR',
            unitPrice: 22.95,
            unitPriceStriked: 25.95,
          },
        }),
      },
    }),
  };
});

// eslint-disable-next-line require-jsdoc
const MockComponent = () => null;

describe('engage > core > hocs > withCurrentProduct', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should inject the context properties into the component', () => {
    const ComposedComponent = withCurrentProduct(MockComponent);
    const wrapper = mount(<ComposedComponent someProp />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(MockComponent).props()).toEqual({
      someProp: true,
      id: '123',
      name: 'acme',
      price: {
        currency: 'EUR',
        unitPrice: 22.95,
        unitPriceStriked: 25.95,
      },
    });
  });

  it('should inject a single property with the context into the component', () => {
    const ComposedComponent = withCurrentProduct(MockComponent, { prop: 'product' });
    const wrapper = mount(<ComposedComponent someProp />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(MockComponent).props()).toEqual({
      someProp: true,
      product: {
        id: '123',
        name: 'acme',
        price: {
          currency: 'EUR',
          unitPrice: 22.95,
          unitPriceStriked: 25.95,
        },
      },
    });
  });
});
