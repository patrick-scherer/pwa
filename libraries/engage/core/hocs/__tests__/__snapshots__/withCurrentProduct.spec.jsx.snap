// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`engage > core > hocs > withCurrentProduct should inject a single property with the context into the component 1`] = `
<WithCurrentProduct(MockComponent)
  someProp={true}
>
  <MockComponent
    product={
      Object {
        "id": "123",
        "name": "acme",
        "price": Object {
          "currency": "EUR",
          "unitPrice": 22.95,
          "unitPriceStriked": 25.95,
        },
      }
    }
    someProp={true}
  />
</WithCurrentProduct(MockComponent)>
`;

exports[`engage > core > hocs > withCurrentProduct should inject the context properties into the component 1`] = `
<WithCurrentProduct(MockComponent)
  someProp={true}
>
  <MockComponent
    id="123"
    name="acme"
    price={
      Object {
        "currency": "EUR",
        "unitPrice": 22.95,
        "unitPriceStriked": 25.95,
      }
    }
    someProp={true}
  />
</WithCurrentProduct(MockComponent)>
`;
